### PR TITLE
Fixed invalid index id

### PR DESCRIPTION
**Pull request by iurii-i-popov**
Fixed late materialization with several indexes when one of them is primary.

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9443/

### CODE REVIEW

File: arangod/Aql/IndexNodeOptimizerRules.cpp [hunk 1]
```diff
@@ -45,13 +45,13 @@ namespace {
...
         // use one index only
-        if (commonIndexId != 0 && commonIndexId != indexId) {
+        if (commonIndexId != std::numeric_limits<TRI_idx_iid_t>::max() && commonIndexId != indexId) {
           continue;
         }
         size_t indexFieldNum = 0;
         for (auto const& field : index->fields()) {
           if (arangodb::basics::AttributeName::isIdentical(nodeAttr.attr, field, false)) {
-            if (commonIndexId == 0) {
+            if (std::numeric_limits<TRI_idx_iid_t>::max() == commonIndexId) {
```
Comment (joerg84) [inline arangod/Aql/IndexNodeOptimizerRules.cpp:54]: Could you switch order for consistentcy
 commonIndexId == std::numeric_limits<TRI_idx_iid_t>::max()

Comment (iurii-i-popov) [reply arangod/Aql/IndexNodeOptimizerRules.cpp:54]: I used https://en.wikipedia.org/wiki/Yoda_conditions

Comment (iurii-i-popov) [reply arangod/Aql/IndexNodeOptimizerRules.cpp:54]: It will be as commonIndexId.isNone() in devel later.

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-mac-onlycatch/7179/

**Comment by KVS85** [comment]
Tests blue except unrelated failures.

**Pull request merged**